src/pages/home/components/mainPanel/child/saleTicket/SaleTicket.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * Блок - Кассир
+ * Элемент - Продажа билетов
+ */
+
+import { useSearchParams } from "react-router-dom";
+import { useMainStore } from "../../MainPanel.store";
+import { Button, Stacks, Step, Stepper } from "@/shared/ui";
+import { useStepper } from "@/shared/hooks";
+import { BusIcon, PaymentIcon, ReturnIcon, SeatIcon, UserIcon } from "@/shared/assets";
+import { WayMenu, WayMainList, SeatMainItem, PassengerInfoItem } from "./index";
+
+export const SaleTicket = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const step = searchParams.get("step") || "0";
+
+  const {
+    saleTicket: { activeWay, way, wayDetails, passengers },
+  } = useMainStore();
+
+  const steps: Step[] = [
+    { icon: <BusIcon /> },
+    { icon: <SeatIcon /> },
+    { icon: <UserIcon /> },
+    { icon: <ReturnIcon />, visible: Boolean(way.returnHave) },
+    { icon: <PaymentIcon /> },
+  ];
+
+  const { activeStep, nextStep, prevStep, isFirstStep, isLastStep } = useStepper({
+    initialStep: Number(step),
+    steps,
+  });
+
+  const handleNextStep = () => {
+    if (activeStep === 0 && !activeWay) {
+      alert("Выберите маршрут");
+      return;
+    } else if (activeStep === 1 && activeWay?.seatsSelected.length === 0) {
+      alert("Выберите места");
+      return;
+    } else if (activeStep === 2 && passengers.some((p) => !p.ticket)) {
+      alert("Заполните данные пассажира");
+      return;
+    } else {
+      setSearchParams({
+        step: String(activeStep + 1),
+      });
+      nextStep();
+    }
+  };
```

Comment:
может будет избегать else if?

Reply:
Какие есть предложения?

Reply:
Можно через switch case 

```suggestion
 const handleNextStep = () => {
  switch (true) {
    case activeStep === 0 && !activeWay:
      alert("Выберите маршрут");
      return;
    case activeStep === 1 && activeWay?.seatsSelected.length === 0:
      alert("Выберите места");
      return;
    case activeStep === 2 && passengers.some((p) => !p.ticket):
      alert("Заполните данные пассажира");
      return;
    default:
      setSearchParams({
        step: String(activeStep + 1),
      });
      nextStep();
  }
};
```